main_test.go

```diff
@@ -149,6 +149,114 @@ func Test_findSecrets(t *testing.T) {
 	}
 }
 
+func Test_findAllSecrets(t *testing.T) {
+	type args struct {
+		getter func(input *ssm.GetParametersByPathInput) (*ssm.GetParametersByPathOutput, error)
+		ns     string
+	}
+	sharedGetter := func(input *ssm.GetParametersByPathInput) (*ssm.GetParametersByPathOutput, error) {
+		if *input.Path == "/prefix/1/" {
+			if input.NextToken == nil {
+				return &ssm.GetParametersByPathOutput{
+					NextToken: aws.String("2"),
+					Parameters: []*ssm.Parameter{
+						{
+							Name:  aws.String("/prefix/1/VALUE_FIRST_ONLY"),
+							Value: aws.String("I CAME FROM PREFIX 1"),
+						},
+					},
+				}, nil
+			} else {
+				return &ssm.GetParametersByPathOutput{
+					NextToken: nil,
+					Parameters: []*ssm.Parameter{
+						{
+							Name:  aws.String("/prefix/1/VALUE_IN_BOTH"),
+							Value: aws.String("I CAME FROM PREFIX 1"),
+						},
+					},
+				}, nil
+			}
+		}
+		if *input.Path == "/prefix/2/" {
+			if input.NextToken == nil {
+				return &ssm.GetParametersByPathOutput{
+					NextToken: aws.String("2"),
+					Parameters: []*ssm.Parameter{
+						{
+							Name:  aws.String("/prefix/2/VALUE_SECOND_ONLY"),
+							Value: aws.String("I CAME FROM PREFIX 2"),
+						},
+					},
+				}, nil
+			} else {
+				return &ssm.GetParametersByPathOutput{
+					NextToken: nil,
+					Parameters: []*ssm.Parameter{
+						{
+							Name:  aws.String("/prefix/2/VALUE_IN_BOTH"),
+							Value: aws.String("I CAME FROM PREFIX 2"),
+						},
+					},
+				}, nil
+			}
+		}
+		// these are returned when empty prefix is sent.
+		return &ssm.GetParametersByPathOutput{
+			NextToken: nil,
+			Parameters: []*ssm.Parameter{
+			},
+		}, nil
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    []string
+		wantErr bool
```

Comment:
Since `findAllSecrets` doesn't return an error, we can safely omit `wantErr` from this struct.